modules/core/src/webgl/utils/query-manager.js

```diff
@@ -51,26 +49,30 @@ class QueryManager {
   }
 
   // Starts a query, sets up a new promise
-  beginQuery(query, onComplete = noop, onError = noop) {
+  beginQuery(query, onComplete, onError) {
     // Make sure disjoint state is cleared, so that this query starts fresh
     // Cancel other queries if needed
     this.cancelInvalidQueries(query.gl);
 
     // Cancel current promise - noop if already resolved or rejected
     this.cancelQuery(query);
 
-    // Create a new promise with attached resolve and reject methods
-    const resolvers = {};
-    query.promise = new Promise((resolve, reject) => {
-      resolvers.resolve = resolve;
-      resolvers.reject = reject;
-    });
-    Object.assign(query.promise, resolvers);
-
     // Add this query to the pending queries
     this.pendingQueries.add(query);
-    // Register the callbacks
-    return query.promise.then(onComplete).catch(onError);
+
+    // Create a new promise with attached resolve and reject methods
+
+    if (onComplete || onError) {
+      const resolvers = {};
+      query.promise = new Promise((resolve, reject) => {
+        resolvers.resolve = resolve;
+        resolvers.reject = reject;
+      });
+      Object.assign(query.promise, resolvers);
+
+      // Register the callbacks
+      query.promise.then(onComplete).catch(onError);
```

Comment:
> query.promise.then(onComplete).catch(onError);

One of the callbacks could be null here, right? Are `null` callbacks ok, if not maybe write `onComplete || x => x, onError || x => x`?

Reply:
I didn't consider this but it looks like it's ok. From the MDN docs for [then](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then):
> If one or both arguments are omitted or are provided non-functions, then then will be missing the handler(s), but will not generate any errors. If the Promise that then is called on adopts a state (fulfillment or rejection) for which then has no handler, a new Promise is created with no additional handlers, simply adopting the final state of the original Promise on which then was called.